draft-ietf-quic-ack-frequency.md

```diff
@@ -335,17 +330,32 @@ send an acknowledgement immediately on receiving a reordered ack-eliciting
 packet. This extension modifies this behavior.
 
 If the endpoint has not yet received an ACK_FREQUENCY frame, or if the most
-recent frame received from the peer has an `Ignore Order` value of `false`
-(0x00), the endpoint MUST immediately acknowledge any subsequent packets that
+recent frame received from the peer has a `Reordering Threshold` value that is
+not 0x00, the endpoint MUST immediately acknowledge any subsequent packets that
 are received out of order.
 
-If the most recent ACK_FREQUENCY frame received from the peer has an `Ignore
-Order` value of `true` (0x01), the endpoint does not make this exception. That
+If the most recent ACK_FREQUENCY frame received from the peer has a `Reordering
+Threshold` value of 0x00, the endpoint does not make this exception. That
 is, the endpoint MUST NOT send an immediate acknowledgement in response to
 packets received out of order, and instead continues to use the peer's
 `Ack-Eliciting Threshold` and `max_ack_delay` thresholds for sending
 acknowledgements.
 
+### Reordering Threshold
+
+If the most recent value of 'Reordering Threshold' is not 0x00, then in addition
+to the immediate acknowledgement when reordering is detected, an additional
```

Comment:
I would recommend splitting this sentence, the bolting on of the "an additional ..." makes it hard to interpret. If I understand this correctly the main part of the sentence is to send an immediate ack when reordering of the threshold value or greater has been detected. The second part is more a clarification that even if a reordering immediate ack has been sent if a lost packet detected event occurs for the same packet then an another immediate ack would be sent. I think that might be clear to reformulate it.

In addition what conditions makes a receiver upgrade a reorder threshold exceeded to lost detected, isn't this the same? I thought in reality the only thing that can occur is that a reorder can go from packet missing (due to reorder) to acknowledged. If it is lost it will never show up and the gap persists.

Reply:
Yes, there is not really any loss detection as the receiver side. The only question is when to send an ack. There are two cases: the normal delayed-ack timer or the re-ordering threshold.